lib/src/r_tree/node.dart

```diff
@@ -20,33 +20,33 @@ part of r_tree;
 /// abstract class, see [LeafNode] and [NonLeafNode] for more information.
 abstract class Node<E> extends RTreeContributor {
   /// The branch factor this node is configured with, which determines when the node should split
-  final int branchFactor;
+  final int? branchFactor;
 
   /// Parent node of this node, or null if this is the root node
-  Node<E> parent;
+  Node<E>? parent;
 
-  Rectangle _minimumBoundingRect;
+  Rectangle? _minimumBoundingRect;
 
   /// Returns the rectangle this Node covers
-  Rectangle get rect => _minimumBoundingRect;
+  Rectangle? get rect => _minimumBoundingRect;
 
   Node(this.branchFactor);
 
   /// Returns an iterable of all items within [searchRect]
-  Iterable<RTreeDatum<E>> search(
-      Rectangle searchRect, bool Function(E item) shouldInclude);
+  Iterable<RTreeDatum<E>?> search(
+      Rectangle? searchRect, bool Function(E item) shouldInclude);
 
   /// Inserts [item] into this node
-  Node<E> insert(RTreeDatum<E> item);
+  Node<E>? insert(RTreeDatum<E>? item);
 
   /// Removes [item] from this node
-  remove(RTreeDatum<E> item);
+  remove(RTreeDatum<E>? item);
 
   /// Remove all children from this node
   clearChildren();
 
   /// Returns a list of all items in this node
-  List<RTreeContributor> get children;
+  List<RTreeContributor?> get children;
```

Comment:
This feels like a place where it would be nice to know you're always going to get a non-null list ... and if there are no children, you get an empty list.